src/nwbinspector/_nwb_inspection.py

```diff
@@ -145,8 +144,17 @@ def inspect_all(
     identifiers = defaultdict(list)
     for nwbfile_path in nwbfiles:
         with pynwb.NWBHDF5IO(path=nwbfile_path, mode="r", load_namespaces=True) as io:
-            nwbfile = robust_s3_read(io.read)
-            identifiers[nwbfile.identifier].append(nwbfile_path)
+            try:
+                nwbfile = io.read()
+                identifiers[nwbfile.identifier].append(nwbfile_path)
+            except Exception as exception:
+                yield InspectorMessage(
+                    message=traceback.format_exc(),
+                    importance=Importance.ERROR,
```

Comment:
Also adding this in here now - it's a known problem that this pre-check can fail to collect and report messages if is read error occurs

Very hard to add tests for ATM though, might try in a follow-up that improves testing all around

Reply:
Sounds good. To clarify, at this point streaming is being redirected to `inspect_dandiset` earlier in `inspect_all`, so this except here is to catch local read errors?

Reply:
Well, the check itself is a special one looking for duplicated file identifiers. That can't be done in an easy way as a normal check function, so it is done here before main deployment of the check registry

And yes, only applies locally to folder contents. Have not implemented a Dandiset way of doing it yet

Reply:
Ah ok got it, thanks for explaining!